app/components/root/root.js

```diff
@@ -29,21 +40,98 @@ export default class Root extends React.Component {
     componentDidMount() {
         AppState.addEventListener('change', this.handleAppStateChange);
         EventEmitter.on(Constants.CONFIG_CHANGED, this.handleConfigChanged);
+        MessageBarManager.registerMessageBar(this.refs.notification);
+        this.configurePushNotifications();
     }
 
     componentWillUnmount() {
         AppState.removeEventListener('change', this.handleAppStateChange);
         EventEmitter.off(Constants.CONFIG_CHANGED, this.handleConfigChanged);
+        MessageBarManager.unregisterMessageBar();
+        PushNotification.unregister();
     }
 
+    onRegisterDevice = (data) => {
+        const prefix = Platform.OS === 'ios' ? Constants.PUSH_NOTIFY_APPLE_REACT_NATIVE : Constants.PUSH_NOTIFY_ANDROID_REACT_NATIVE;
+        this.props.actions.setDeviceToken(`${prefix}:${data.token}`);
+    };
+
+    onPushNotification = (notification) => {
+        const {foreground, userInteraction, data, message} = notification;
+        let noty;
+
+        if (Platform.OS === 'android') {
+            noty = {
+                data: {
+                    channel_id: notification.channel_id,
+                    team_id: notification.team_id
+                },
+                message
+            };
+        } else {
+            noty = {
+                data,
+                message
+            };
+        }
+
+        if (foreground) {
+            this.handleInAppNotification(noty);
+        } else if (userInteraction) {
+            this.onNotificationTapped(noty);
+        }
```

Comment:
No, `foreground` is true when the app is open and you receive a push notification. `userInteraction` is true when the user has actually tapped on the push notification